File: aspnet-core/Klir.TechChallenge.Web.Api/Entities/ShoppingCart.cs
using System.Collections.Generic;
using Klir.TechChallenge.Web.Api.DTOs;

namespace Klir.TechChallenge.Web.Api.Entities
{
    public class ShoppingCart
    {
        public int Id { get; set; }
        public List<CartProductDto> CartProductsDto { get; set; }
        public int Quantidy { get; set; }
        public decimal TotalPrice { get; set; }
        public decimal OriginalPrice { get; set; }
        public decimal Saved { get; set; }
    }
}

File: angular/src/app/_models/shopping-cart.ts
import { CartProductDto } from "../_DTOs/cart-produtc-dto";
import { CartProduct } from "./cart-product";

export interface ShoppingCart {
  cartProductDtos: CartProductDto[];
  totalPrice?: number;
  saved?: number;
  originalPrice?: number;
  quantidy?: number;
}
